<!DOCTYPE html>
<html lang="en">
    <head>
        <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
        <title>fnl.es &middot; Projects</title>
        <link rel="shortcut icon" href="http://fnl.es/favicon.ico" />
<link href="http://fnl.es/feeds/all.atom.xml" type="application/atom+xml" rel="alternate" title="fnl.es Atom Feed" />
        <link href="http://fnl.es/feeds/all.rss.xml" type="application/rss+xml" rel="alternate" title="fnl.es RSS Feed" />

        <link rel="stylesheet" href="http://fnl.es/theme/css/screen.css" type="text/css" />
        <link rel="stylesheet" href="http://fnl.es/theme/css/pygments.css" type="text/css" />
        <!-- Start of StatCounter Code for Default Guide -->
<script type="text/javascript">
var sc_project=3728421; 
var sc_invisible=1; 
var sc_security="a5624e69"; 
</script>
<script type="text/javascript"
src="http://www.statcounter.com/counter/counter.js"></script>
        <!-- End of StatCounter Code for Default Guide -->
    </head>
    <body>
        <!-- Start of StatCounter Code for Default Guide -->
<noscript><div class="statcounter"><a title="hit counter"
href="http://statcounter.com/free-hit-counter/"
target="_blank"><img class="statcounter"
src="http://c.statcounter.com/3728421/0/a5624e69/1/"
alt="hit counter"></a></div></noscript>
        <!-- End of StatCounter Code for Default Guide -->
<div id="header">
            <ul id="nav">
                <li><a class="flagred" href="http://fnl.es">Home</a></li>
<li><a class="flagyellow" href="http://fnl.es/pages/about.html">About</a></li>
<li class="selected"><a class="flagyellow" href="http://fnl.es/pages/projects.html">Projects</a></li>
<li><a class="flagred" href="http://fnl.es/archives.html">Archives</a></li>
            </ul>
            <div class="header_box">
                <h1><a href="http://fnl.es">fnl.es</a></h1>
                <h2><a href="mailto:flo@fnl.es" id="webmaster">fnl</a> en Espa√±a</h2>            </div>
        </div>
        <div id="wrapper">
            <div id="content">
                <div class="page">
                    <h1>Projects</h1>
                    <p><strong>Text Mining in Biomedicine</strong></p>
<p><a class="reference external" href="https://github.com/fnl/txtfnnl">txtfnnl</a></p>
<blockquote>
My current main project, a modular text mining pipeline for biomedicine using the UIMA framework.
It provides a set of Annotation Engine (AE) modules that can be combined to run more complex text
mining tasks such as Gene Normalization (assigning DB IDs to gene symbols) or mining for
biological events (e.g., transcription regulation interactions). My framework adds a special
builder pattern implementation to make UIMA/uimaFIT AEs and Resources less of a chore to develop,
maintain, and configure. In addition, my framework provides a &quot;type-free&quot; &quot;one-model-fits-all&quot;
type system, that is a single text annotation type with an <tt class="docutils literal">annotator&#64;namespace:identifier</tt>
schema to make any kind of annotations on a UIMA SOFA. Last it has special wrappers for the
Apache Tika project (a libary to extract text content from many different file-types) that add
functionality beyond the raw extraction alone.</blockquote>
<p><a class="reference external" href="https://pypi.python.org/pypi/medic/">medic</a></p>
<blockquote>
A command-line tool to manage an up-to-date PubMed/MEDLINE DB mirror. This is a Python3 program
that allows you to bootstrap a MEDLINE Citation (PubMed Abstracts) repository in either SQLite or
PostgreSQL (Technically, it uses SQL Alchemy, so most datastores should be OK.) You can import
data directly from NCBI eUtils or by parsing MEDLINE XML files. Content can be extracted in
plain-text (full, TIAB only, or as TSV) or HTML formats. This is an evolution of the <a class="reference external" href="https://github.com/fnl/libfnl">libfnl</a>
<tt class="docutils literal">medline</tt> tool. The easiest way to install it is from PyPi, using &quot;<tt class="docutils literal">pip3 install medic</tt>&quot;.
To use it, check out &quot;<tt class="docutils literal">medic <span class="pre">--help</span></tt>&quot; and &quot;<tt class="docutils literal">man medic</tt>&quot;.</blockquote>
<p><a class="reference external" href="https://github.com/fnl/gnamed">gnamed</a></p>
<blockquote>
A command-line tool to maintain an up-to-date database of gene and protein names together with
their most important keywords (kDa, length, chromosome position, etc.) and their references to
the NCBI taxonomy and to PubMed. This is very useful to build gene normalization systems.
Beyond just duming the data, this tool groups unique genes and proteins from different databases
and links proteins and genes (as a <tt class="docutils literal">n to m</tt> relation).
To use it, clone it from GitHub (&quot;<tt class="docutils literal">git clone git&#64;github.com:fnl/gnamed.git</tt>&quot;) and install it
locally (&quot;<tt class="docutils literal"><span class="pre">virtualenv-3.3</span> gnamed; cd gnamed; . bin/activate; python3 setup.py install</tt>&quot;).
Then, you can learn about it by using <tt class="docutils literal">man gnamed</tt> (or just visit the link...).</blockquote>
<p><a class="reference external" href="https://github.com/fnl/libfnl">libfnl</a></p>
<blockquote>
My continuous Python library for supporting all types of BioNLP tasks I am interested in:
<tt class="docutils literal">gnamed</tt> is a library to import gene and protein symbols, names, keywords, and references into
a unified PostgreSQL repository. <tt class="docutils literal">medline</tt> is a tool to maintain a on-site instance of the
NLM MEDLINE database. Several other minor tools and libraries are available, such as annotator
evaluation or interfacing to CouchDB with my own Python 3k client (written a few years ago when
there was no other Py3k CouchDB client around). Essentially it is the &quot;source&quot; of the two
projects just described above.</blockquote>
<p><a class="reference external" href="http://www.biocreative.org">BioCreative</a></p>
<blockquote>
An international project organizing community challenges in text mining for molecular biology.
I have in particular been the main organizer of one of these challenges, BioCreative II.5, and
have developed a Django-based website to manage these events and their participants.
Also part of this project is my <a class="reference external" href="https://github.com/fnl/bceval">bceval</a> evaluation script to evaluate the perfomance of text
mining systems.</blockquote>
<p><strong>Sidekicks and Minor Projects</strong></p>
<p><a class="reference external" href="https://github.com/fnl/couchdb_client">couchdb_client</a></p>
<blockquote>
A tiny and efficient JavaScript/Node.js client for CouchDB.</blockquote>
<p><a class="reference external" href="https://github.com/fnl/libfsmg">libfsmg</a></p>
<blockquote>
A generic finite state machine library in Java.
Design your own grammar to quickly build generic pattern matching engines.</blockquote>
<p><a class="reference external" href="https://github.com/fnl/sentence_splitter">sentence_splitter</a></p>
<blockquote>
A simple command-line tool to quickly split sentences based on regular patterns.</blockquote>
<p><a class="reference external" href="https://github.com/fnl/tred">tred</a></p>
<blockquote>
Load (parts of) the <a class="reference external" href="http://rulai.cshl.edu/cgi-bin/TRED/tred.cgi?process=home">Transcriptional Regulatory Element Database</a> into a PostgreSQL DB.</blockquote>

                </div>

                <div class="clear"></div>
                <div id="footer">
                    <p>
                    <a class="atom" href="http://fnl.es/feeds/all.atom.xml">RSS Feed</a>
                    </p>
                </div>
            </div>
            <div class="clear"></div>
        </div>
    </body>
</html>